core/variables.js

```diff
@@ -267,11 +268,9 @@ Blockly.Variables.getUses = function(name, workspace) {
 
 /**
  * When a variable is deleted, find and dispose of all uses of it.
- * @param {string} name Name of deleted variable.
- * @param {!Blockly.Workspace} workspace The workspace to delete uses from.
+ * @param {!Array.<!Blockly.Block>} uses An array of blocks using the variable.
  */
-Blockly.Variables.disposeUses = function(name, workspace) {
-  var uses = Blockly.Variables.getUses(name, workspace);
+Blockly.Variables.disposeUses = function(uses) {
```

Comment:
I think the old API was a bit better for working with. It let developers call it without figuring out how to get the full list of blocks using it. Maybe make it just a lightweight wrapper that calls a private version that takes the list of uses so you can do the optimization below?